installer/deployment.go

```diff
@@ -121,6 +121,80 @@ func newDeployment(authopts AuthOptions, authzopts AuthzOptions) (objects []runt
 			Effect:   core.TaintEffectNoSchedule,
 		})
 	}
+	if authopts.HttpsProxy != "" || authopts.HttpProxy != "" || authopts.NoProxy != "" {
+		proxyEnvVarRef := core.EnvFromSource{
+			SecretRef: &core.SecretEnvSource{},
+		}
+
+		proxyEnvVarRef.SecretRef.Name = "guard-proxy"
+		d.Spec.Template.Spec.Containers[0].EnvFrom = append(d.Spec.Template.Spec.Containers[0].EnvFrom, proxyEnvVarRef)
+		if authopts.ProxyCert != "" {
+			proxycertVolumeRef := core.Volume{
+				Name: "proxy-certstore",
+				VolumeSource: core.VolumeSource{
+					Secret: &core.SecretVolumeSource{
+						SecretName: "guard-proxy-cert",
+					},
+				},
+			}
+
+			sslCertsVolumeRef := core.Volume{
+				Name: "ssl-certs",
+				VolumeSource: core.VolumeSource{
+					EmptyDir: &core.EmptyDirVolumeSource{},
+				},
+			}
+
+			d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes, proxycertVolumeRef)
+			d.Spec.Template.Spec.Volumes = append(d.Spec.Template.Spec.Volumes, sslCertsVolumeRef)
+
+			sslCertsVolumeMount := core.VolumeMount{
+				Name:      "ssl-certs",
+				MountPath: "/etc/ssl/certs/",
+			}
+
+			d.Spec.Template.Spec.Containers[0].VolumeMounts = append(d.Spec.Template.Spec.Containers[0].VolumeMounts, sslCertsVolumeMount)
+
+			optionalSecret := false
+			initContainer := []core.Container{
+				{
+					Name:  "add-certs",
```

Comment:
will this be displayed in kubectl get pod. If yes, we should have better name.

Reply:
changed to update-proxy-certs